pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -209,6 +209,22 @@ self: super: {
   ### END HASKELL-LANGUAGE-SERVER SECTION ###
   ###########################################
 
+  # basement is broken on some 32-bit systems due to missing imports
+  # See: https://github.com/haskell-foundation/foundation/issues/565
+  # This patch is taken from a PR to fix the above issue. See:
+  # https://github.com/haskell-foundation/foundation/issues/573
+  # However, that PR doesn't work completely, because it doesn't apply to
+  # all the GHC versions we want to fix. So the actual patch file is modified
+  # to also affect those other versions.
+  #
+  # The patch applies, but does not compile, on non-32-bit systems.
+  #
+  # This patch can be removed once the above issue is closed and the fix is
+  # available on Hackage (and Nixpkgs.)
+  basement = if pkgs.stdenv.isi686
```

Comment:
In theory, are there i686 systems where this patch shouldn't apply?  Like, non-linux?  I'm not very familiar with all the other architectures that nixpkgs commonly supports.

Reply:
Frankly, I'm not sure either. I believe that this patch should actually apply on all `i686` systems, since the compilation issue is about a module which is exposed based on system word size (and that's OS-independent.)

Perhaps the patch should apply to all 32-bit architectures, and not just `i686`, but
  1. I don't know if any others are supported
  2. I wouldn't be able to test them if they were

Reply:
This patch should be applied globally if anything, platform specific patches are a nightmare to maintain and should be avoided. Any conditionals that are needed should take advantage of platform conditionals via CPP.

It seemed from upstream discussion that the problem affects all 32 bit platforms including e.g. ghcjs.